File: Programming Fundamentals with Python/Data Types and Variables/water_overflow.py

n = int(input())

water_tank = 255
capacity = 0
for i in range(n):
    liters_of_water = int(input())

    if water_tank - liters_of_water < 0:
        print("Insufficient capacity!")
        continue
    water_tank -= liters_of_water
    capacity += liters_of_water
print(capacity)

File: Programming Fundamentals with Java/Data Types and Variables/WaterOverflow.java
import java.util.Scanner;

public class WaterOverflow 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int n = Integer.parseInt(input.nextLine());
        
        int capacity = 0;
        for (int i = 0; i < n; i++)
        {
            int num = Integer.parseInt(input.nextLine());
            capacity += num;
            if (capacity > 255)
            {
                capacity -= num;
                System.out.println("Insufficient capacity!");
            }
        }

        System.out.println(capacity);
    }
}
